src/main/java/org/opentripplanner/routing/algorithm/raptor/router/TransitRouter.java

```diff
@@ -0,0 +1,273 @@
+package org.opentripplanner.routing.algorithm.raptor.router;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.opentripplanner.model.plan.Itinerary;
+import org.opentripplanner.routing.algorithm.mapping.RaptorPathToItineraryMapper;
+import org.opentripplanner.routing.algorithm.raptor.router.street.AccessEgressRouter;
+import org.opentripplanner.routing.algorithm.raptor.router.street.FlexAccessEgressRouter;
+import org.opentripplanner.routing.algorithm.raptor.transit.AccessEgress;
+import org.opentripplanner.routing.algorithm.raptor.transit.Transfer;
+import org.opentripplanner.routing.algorithm.raptor.transit.TransitLayer;
+import org.opentripplanner.routing.algorithm.raptor.transit.TripSchedule;
+import org.opentripplanner.routing.algorithm.raptor.transit.mappers.AccessEgressMapper;
+import org.opentripplanner.routing.algorithm.raptor.transit.mappers.RaptorRequestMapper;
+import org.opentripplanner.routing.algorithm.raptor.transit.request.RaptorRoutingRequestTransitData;
+import org.opentripplanner.routing.algorithm.raptor.transit.request.RoutingRequestTransitDataProviderFilter;
+import org.opentripplanner.routing.algorithm.raptor.transit.request.TransitDataProviderFilter;
+import org.opentripplanner.routing.algorithm.transferoptimization.configure.TransferOptimizationServiceConfigurator;
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.routing.api.request.StreetMode;
+import org.opentripplanner.routing.api.response.InputField;
+import org.opentripplanner.routing.api.response.RoutingError;
+import org.opentripplanner.routing.api.response.RoutingErrorCode;
+import org.opentripplanner.routing.error.RoutingValidationException;
+import org.opentripplanner.routing.framework.DebugTimingAggregator;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graph.GraphIndex;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.services.FareService;
+import org.opentripplanner.standalone.server.Router;
+import org.opentripplanner.transit.raptor.RaptorService;
+import org.opentripplanner.transit.raptor.api.path.Path;
+import org.opentripplanner.transit.raptor.api.request.SearchParams;
+import org.opentripplanner.transit.raptor.api.response.RaptorResponse;
+import org.opentripplanner.util.OTPFeature;
+
+public class TransitRouter {
+
+    public static final int NOT_SET = -1;
+
+    public static TransitRouterResult route(
+            RoutingRequest request,
+            Router router,
+            DebugTimingAggregator debugTimingAggregator
```

Comment:
Consider making these members of the class. If the request is changed it should be a parameter on the method, if it is read-only then I think it make sense to have it as a member. This class have only static methods, but is not final and is missing a `private` constructor. I think we should be consistent, either this is a pure utility class(static) or we should create an instance - I think the last is the best here.